lib/spack/spack/util/s3.py

```diff
@@ -4,83 +4,115 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 import os
 import urllib.parse
+from typing import Any, Dict, Tuple
 
 import spack
+import spack.config
 import spack.util.url as url_util
 
+#: Map (mirror name, method) tuples to s3 client instances.
```

Comment:
Based on our conversation in slack, I was expecting a mapping from `(url, method)` to s3 client instance, rather than `(name, method)` to client instance.